pages_domain.go

```diff
@@ -0,0 +1,173 @@
+
+	var pageDomainResponse PagesDomainsResponse
+	if err := json.Unmarshal(res, &pageDomainResponse); err != nil {
+		return []PagesDomain{}, err
+	}
+	return pageDomainResponse.Result, nil
+}
+
+// GetPagesDomain gets a single domain.
+//
+// API Reference: https://api.cloudflare.com/#pages-domains-get-domains
+func (api *API) GetPagesDomain(ctx context.Context, params PagesDomainParameters) (PagesDomain, error) {
+	if params.AccountID == "" {
+		return PagesDomain{}, ErrMissingAccountID
+	}
+
+	if params.ProjectName == "" {
+		return PagesDomain{}, ErrMissingProjectName
+	}
+
+	if params.DomainName == "" {
+		return PagesDomain{}, ErrMissingDomain
+	}
+
+	uri := fmt.Sprintf("/accounts/%s/pages/projects/%s/domains/%s", params.AccountID, params.ProjectName, params.DomainName)
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return PagesDomain{}, err
+	}
+
+	var pagesDomainResponse PagesDomainResponse
+	if err := json.Unmarshal(res, &pagesDomainResponse); err != nil {
+		return PagesDomain{}, err
+	}
+	return pagesDomainResponse.Result, nil
+}
+
+// PagesPatchDomain retries the validation status of a single domain.
+//
+// API Reference: https://api.cloudflare.com/#pages-domains-patch-domain
+func (api *API) PagesPatchDomain(ctx context.Context, params PagesDomainParameters) (PagesDomain, error) {
+	if params.AccountID == "" {
+		return PagesDomain{}, ErrMissingAccountID
+	}
+
+	if params.ProjectName == "" {
+		return PagesDomain{}, ErrMissingProjectName
+	}
+
+	if params.DomainName == "" {
+		return PagesDomain{}, ErrMissingDomain
+	}
+
+	uri := fmt.Sprintf("/accounts/%s/pages/projects/%s/domains/%s", params.AccountID, params.ProjectName, params.DomainName)
+	res, err := api.makeRequestContext(ctx, http.MethodPatch, uri, nil)
+	if err != nil {
+		return PagesDomain{}, err
+	}
+
+	var pagesDomainResponse PagesDomainResponse
+	if err := json.Unmarshal(res, &pagesDomainResponse); err != nil {
+		return PagesDomain{}, err
+	}
+	return pagesDomainResponse.Result, nil
+}
+
+// PagesAddDomain adds a domain to a pages project.
+//
+// API Reference: https://community.cloudflare.com/t/add-pages-domain-via-api/303510/3
+// Blame Washly for not making it official.
+func (api *API) PagesAddDomain(ctx context.Context, params PagesDomainParameters) (PagesDomain, error) {
+	if params.AccountID == "" {
+		return PagesDomain{}, ErrMissingAccountID
+	}
+
+	if params.ProjectName == "" {
+		return PagesDomain{}, ErrMissingProjectName
+	}
+
+	if params.DomainName == "" {
+		return PagesDomain{}, ErrMissingDomain
+	}
+
+	uri := fmt.Sprintf("/accounts/%s/pages/projects/%s/domain", params.AccountID, params.ProjectName)
+	res, err := api.makeRequestContext(ctx, http.MethodPost, uri, params)
+	if err != nil {
+		return PagesDomain{}, err
+	}
+
+	var pagesDomainResponse PagesDomainResponse
+	if err := json.Unmarshal(res, &pagesDomainResponse); err != nil {
+		return PagesDomain{}, err
+	}
+	return pagesDomainResponse.Result, nil
+}
```

Comment:
let's add `Delete` here too; the fragment will be `#pages-domains-delete-domain`

```
DELETE accounts/:account_id/pages/projects/:project_name/domains/:domain_name
```